provider_test.go

```diff
@@ -58,7 +58,7 @@ func TestGetAPI(test *testing.T) {
 	// when the address is an empty string, we get a nullAPI
 	var api client.API
 
-	api, err := getAPI("", 1)
+	api, err := getAPI("null", "endpoint", 1, "etcd_prefix")
```

Comment:
This doesn't seem to be testing "when the address is an empty string, we get a nullAPI", though maybe it doesn't matter anymore. It used to be that if the `tunnel_address` was something like `${aws_instance.coreos.public_ip}`, then when terraform would expand that variable in the planning stage, before that instance existed, it would expand it to an empty string. When that happened, all we needed to do was to not fail, and then when Terraform would get to the apply stage it wouldn't initialize the fleet provider until after the instance was actually created, and by then we'd have the real IP address and could work normally.

However, this doesn't seem to work in more recent versions of Terraform, for reasons I haven't yet fully investigated. https://github.com/hashicorp/terraform/issues/2430 has a bit more context.

Until that's sorted out, perhaps it's best to retain the old behavior.